src/Raven.Server/RavenServerStartup.cs

```diff
@@ -157,23 +159,47 @@ private async Task RequestHandler(HttpContext context)
                 context.Response.StatusCode = (int)HttpStatusCode.OK;
                 context.Response.Headers["Content-Type"] = "application/json; charset=utf-8";
                 context.Response.Headers[Constants.Headers.ServerVersion] = RavenVersionAttribute.Instance.AssemblyVersion;
+                // Check context url AND TW clients
+                var isQueryBool = (IsQuery(context) && TrafficWatchManager.HasRegisteredClients);
```

Comment:
Not sure if this is possible, but maybe it would be better to introduce some kind dedicated context that would wrap the original HttpContext and this new context (if traffic watch is enabled) would 'record' the original body for the traffic watch. Thoughts?